sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -106,7 +106,27 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
   final def execute(): RDD[InternalRow] = {
     RDDOperationScope.withScope(sparkContext, nodeName, false, true) {
       prepare()
-      doExecute()
+
+      if (sqlContext.conf.wholeStageEnabled && supportCodeGen
```

Comment:
I would push back here.  The whole goal of this initial effort is to put some infrastructure in place so that we can iterate quickly moving forward.  Right now its really hard given any query plan to understand what is happening under the covers because the control flow isn't explicitly visible anymore.